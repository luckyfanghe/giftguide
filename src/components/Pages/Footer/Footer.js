import React from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Link } from "react-router-dom";
import { FormControl, Form, Button, Row, Col } from "react-bootstrap";
import { BrowserRouter } from "react-router-dom";
import * as socialLink from "./social_link";

import * as S from "./styled";

class Footer extends React.Component {
  constructor(props) {
    super(props);
  }

  _onClickSocialHandler = type => {
    window.open(type);
  };

  render() {
    return (
      <footer>
        <React.Fragment>
          <S.Body data-testid="Footer">
            <S.Socail>
              <S.Title>CONNECT WITH GIFT GUIDE</S.Title>
              <S.SocialIcon>
                <S.Icon>
                  <img src={socialLink.PINTEREST_ICON} />
                </S.Icon>
                <S.Icon>
                  <img src={socialLink.INSTAGRAM_ICON} />
                </S.Icon>
                <S.Icon>
                  <img src={socialLink.FACEBOOK_ICON} />
                </S.Icon>
                <S.Icon>
                  <img src={socialLink.TWITTER_ICON} />
                </S.Icon>
              </S.SocialIcon>
              <S.SiteLink>
                <S.Link>CONTACT US</S.Link>
                <S.Link>TEAMS OF USE / PRIVACY</S.Link>
                <S.Link>FEEDBACK</S.Link>
              </S.SiteLink>
            </S.Socail>
            <S.InviteContainer>
              <S.Title>INVITE FRIENDS TO GIFT GUIDE</S.Title>
              <S.SubTitle>enter your loved one's email to invite</S.SubTitle>
              <S.Form>
                <S.Input>
                  <FormControl type="text" onKeyDown={this._handleKeyDown} />
                </S.Input>
                <S.Button>INVITE</S.Button>
              </S.Form>
              <S.Copyright>2019 TM GIFT GUIDE LLC</S.Copyright>
            </S.InviteContainer>
          </S.Body>
        </React.Fragment>
      </footer>
    );
  }
}

const actions = dispatch => bindActionCreators({}, dispatch);

function mapStateToProps(state) {
  // return {
  //   user: state.userReducer.user.user,
  //   banner: state.bannerReducer.banner,
  // };
}

export default Footer;
// export default connect(null, actions)(Footer);
