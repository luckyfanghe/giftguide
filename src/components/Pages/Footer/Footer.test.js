import React from "react";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import Footer from "./Footer";

it("Footer renders corrently", () => {
  const { queryByTestId } = render(<Footer />);

  expect(queryByTestId("Footer")).toBeTruthy();
});

describe("Footer", () => {
  test("Footer snapshot renders ", () => {
    const component = renderer.create(<Footer />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
