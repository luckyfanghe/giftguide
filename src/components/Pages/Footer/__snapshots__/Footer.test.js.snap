// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Footer snapshot renders  1`] = `
<footer>
  <div
    className="sc-AykKC cSFNtd"
    data-testid="Footer"
  >
    <div
      className="sc-AykKD eVoXHJ"
    >
      <div
        className="sc-AykKE gsEnyW"
      >
        CONNECT WITH GIFT GUIDE
      </div>
      <div
        className="sc-AykKF iMRYRu"
      >
        <div
          className="sc-AykKG gJaBCL"
        >
          <img
            src="https://static.wixstatic.com/media/8f6f59264a094af0b46e9f6c77dff83e.png/v1/fill/w_35,h_35,al_c,q_85,usm_0.66_1.00_0.01/8f6f59264a094af0b46e9f6c77dff83e.webp"
          />
        </div>
        <div
          className="sc-AykKG gJaBCL"
        >
          <img
            src="https://static.wixstatic.com/media/01c3aff52f2a4dffa526d7a9843d46ea.png/v1/fill/w_35,h_35,al_c,q_85,usm_0.66_1.00_0.01/01c3aff52f2a4dffa526d7a9843d46ea.webp"
          />
        </div>
        <div
          className="sc-AykKG gJaBCL"
        >
          <img
            src="https://static.wixstatic.com/media/0fdef751204647a3bbd7eaa2827ed4f9.png/v1/fill/w_35,h_35,al_c,q_85,usm_0.66_1.00_0.01/0fdef751204647a3bbd7eaa2827ed4f9.webp"
          />
        </div>
        <div
          className="sc-AykKG gJaBCL"
        >
          <img
            src="https://static.wixstatic.com/media/c7d035ba85f6486680c2facedecdcf4d.png/v1/fill/w_35,h_35,al_c,q_85,usm_0.66_1.00_0.01/c7d035ba85f6486680c2facedecdcf4d.webp"
          />
        </div>
      </div>
      <div
        className="sc-AykKH gTJSEi"
      >
        <div
          className="sc-AykKI eIxMid"
        >
          CONTACT US
        </div>
        <div
          className="sc-AykKI eIxMid"
        >
          TEAMS OF USE / PRIVACY
        </div>
        <div
          className="sc-AykKI eIxMid"
        >
          FEEDBACK
        </div>
      </div>
    </div>
    <div
      className="sc-AykKJ hDJrHN"
    >
      <div
        className="sc-AykKE gsEnyW"
      >
        INVITE FRIENDS TO GIFT GUIDE
      </div>
      <div
        className="sc-AykKK dMeBJD"
      >
        enter your loved one's email to invite
      </div>
      <div
        className="sc-fzXfLO bqoHDp"
      >
        <div
          className="sc-fzXfLP fjiGwV"
        >
          <input
            className="form-control"
            type="text"
          />
        </div>
        <button
          className="sc-fzXfLQ eiOJjv"
          type="button"
        >
          INVITE
        </button>
      </div>
      <div
        className="sc-fzXfLR jaafTt"
      >
        2019 TM GIFT GUIDE LLC
      </div>
    </div>
  </div>
</footer>
`;
